index.test-d.ts

```diff
@@ -0,0 +1,16 @@
+import {expectType} from 'tsd-check';
+import pThrottle, {AbortError} from '.';
+
+const throttledUnicorn = pThrottle((i: string) => {
+	return '🦄';
+}, 1, 1000);
+
+const throttledLazyUnicorn = pThrottle((i: string) => {
+	return Promise.resolve('🦄');
+}, 1, 1000);
```

Comment:
Unrelated to this PR, but while I got you here; What do you think about changing limit and interval to be an options-object? I think the above quite unreadable.